test/migrations/2020-01-15-mobile.test.js

```diff
@@ -621,6 +621,93 @@ describe('2020-01-15 mobile migration', () => {
+    await application.deviceInterface.setRawStorageValue(
+      'migrations',
+      JSON.stringify(['anything'])
+    );
+    const operator002 = new SNProtocolOperator002(new SNWebCrypto());
+    const identifier = 'foo';
+    /** Create old version account parameters */
+    const password = 'tar';
+    const accountKey = await operator002.createRootKey(
+      identifier,
+      password
+    );
+    await application.deviceInterface.setRawStorageValue(
+      'auth_params',
+      JSON.stringify(accountKey.keyParams.getPortableValue())
+    );
+    await application.deviceInterface.setRawStorageValue(
+      'user',
+      JSON.stringify({ email: identifier })
+    );
+    expect(accountKey.keyVersion).to.equal(ProtocolVersion.V002);
+    await application.deviceInterface.legacy_setRawKeychainValue({
+      mk: accountKey.masterKey,
+      pw: accountKey.serverPassword,
+      ak: accountKey.dataAuthenticationKey,
+      jwt: 'foo'
+    });
+    /** Create encrypted item and store it in db */
+    const notePayload = Factory.createNotePayload();
+    const noteEncryptionParams = await operator002.generateEncryptedParameters(
+      notePayload,
+      PayloadFormat.EncryptedString,
+      accountKey,
+    );
+    const noteEncryptedPayload = CreateMaxPayloadFromAnyObject(
+      notePayload,
+      noteEncryptionParams
+    );
+    await application.deviceInterface.saveRawDatabasePayload(noteEncryptedPayload, application.identifier);
+
+    /** Run migration */
+    const promptValueReply = (prompts) => {
+      const values = [];
+      for (const prompt of prompts) {
+        if (prompt.validation === ChallengeValidation.None) {
+          values.push(new ChallengeValue(prompt, password));
+        }
+      }
+      return values;
+    };
+    const receiveChallenge = async (challenge) => {
+      application.addChallengeObserver(challenge, {
+        onInvalidValue: (value) => {
+          const values = promptValueReply([value.prompt]);
+          application.submitValuesForChallenge(challenge, values);
+        },
+      });
+      const initialValues = promptValueReply(challenge.prompts);
+      application.submitValuesForChallenge(challenge, initialValues);
+    };
+    await application.prepareForLaunch({
+      receiveChallenge: receiveChallenge,
+    });
+    await application.launch(true);
+
+    const itemsKey = application.itemManager.itemsKeys()[0];
+    expect(itemsKey.keyVersion).to.equal(ProtocolVersion.V002);
+
+    /** Expect note is decrypted */
+    expect(application.itemManager.notes.length).to.equal(1);
+    const retrievedNote = application.itemManager.notes[0];
+    expect(retrievedNote.uuid).to.equal(notePayload.uuid);
+    expect(retrievedNote.content.text).to.equal(notePayload.content.text);
+
+    expect(await application.getUser().email).to.equal(identifier);
+    console.warn('Expecting exception due to deiniting application while trying to renew session');
+    await application.deinit();
```

Comment:
You could wrap that in a try/catch block

Reply:
Hmm tested now, didn't seem to work. Still threw uncaught exception.